Latest Pyvista release broke CICD

As the title. 


```pytb
=========================================================================== FAILURES ===========================================================================
__________________________________________________________________ test_single_glyph[500-UX] ___________________________________________________________________

mapdl = <ansys.mapdl.core.mapdl_grpc.MapdlGrpc object at 0x7f4dcd2e8830>, cleared = None, field = 'UX', magnitude = 500
verify_image_cache = <pytest_pyvista.pytest_pyvista.VerifyImageCache object at 0x7f4da0f08d40>

    @pytest.mark.parametrize(
        "field", ["UX", "UY", "UZ", "FX", "FY", "FZ", "TEMP", "HEAT", "VOLT", "CHRG"]
    )
    @pytest.mark.parametrize("magnitude", [0, 50, 500])
    def test_single_glyph(mapdl, cleared, field, magnitude, verify_image_cache):
        mapdl.et(1, 226)
        mapdl.keyopt(1, 1, 1011)  # Thermal-Piezoelectric
        mapdl.n(1, 0, 0, 0)
    
        if field in [x for group in DISPL_LABELS for x in group]:
            func = getattr(mapdl, "d")
        else:
            func = getattr(mapdl, "f")
    
        func(1, field, magnitude)
    
        if magnitude > 0:
            mapdl.n(2, 1, 0, 0)
            func(2, field, magnitude * 2)
    
        if magnitude > 50:
            mapdl.n(3, 2, 0, 0)
            func(3, field, magnitude * 10)
    
        mapdl.allsel()
    
>       p = mapdl.nplot(
            plot_bc=True,
            # point_size=max(magnitude, 10),
            # render_points_as_spheres=True,
            plot_bc_legend=True,
            plot_bc_labels=True,
            title="",
        )

tests/test_plotting.py:346: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/misc.py:512: in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/misc.py:492: in wrapper
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/mapdl_extended.py:1251: in nplot
    pl.plot([], points, labels, mapdl=self, **kwargs)
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/plotting/visualizer.py:807: in plot
    self.bc_plot(
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/plotting/visualizer.py:558: in bc_plot
    self.bc_nodes_plot(
        def debug_orders(pl, point):
            pl = pl.scene
            pl.show(auto_close=False)
            pl.window_size = (100, 100)
            width, height = pl.window_size
            if pl._picking_right_clicking_observer is None:
                pl.iren._mouse_left_button_press(
                    int(width * point[0]), int(height * point[1])
                )
                pl.iren._mouse_left_button_release(width, height)
            else:
                pl.iren._mouse_right_button_press(
                    int(width * point[0]), int(height * point[1])
                )
                pl.iren._mouse_right_button_release(width, height)
            pl.iren._mouse_move(int(width * point[0]), int(height * point[1]))
    
        mapdl.asel("S", "area", "", 1)
        if selection == "R" or selection == "U":
            point_to_pick = (285 / 1024, 280 / 800)
            mapdl.asel("a", "area", "", 2)
        elif selection == "A":
            point_to_pick = (285 / 1024, 280 / 800)
        else:
            point_to_pick = (0.5, 0.5)
    
>       selected = mapdl.asel(
            selection,
            "P",
            "area",
            _debug=lambda x: debug_orders(x, point=point_to_pick),
            tolerance=0.2,
        )  # Selects node 2

tests/test_plotting.py:872: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/mapdl_core.py:1490: in inner_wrapper
    output = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/mapdl_extended.py:344: in asel
    return wrapped(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/misc.py:604: in wrapper
    vmin = self._enable_picking_entities(
/__t/Python/3.12.13/x64/lib/python3.12/site-packages/ansys/mapdl/core/mapdl_core.py:2929: in _enable_picking_entities
    _debug(pl)
tests/test_plotting.py:876: in <lambda>
    _debug=lambda x: debug_orders(x, point=point_to_pick),
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tests/test_plotting.py:851: in debug_orders
    if pl._picking_right_clicking_observer is None:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pyvista.plotting.plotter.Plotter object at 0x7f4d9d4dff20>, item = '_picking_right_clicking_observer'

    def __getattr__(self, item: str) -> Any:
        """Resolve plotter component plugin lookups on attribute miss.
    
        Before falling through, check whether ``item`` matches a
        pending ``pyvista.plotter_components`` entry point. A match
        triggers a one-shot plugin import, after which normal attribute
        resolution finds the newly-attached component descriptor.
    
        Mirrors :meth:`pyvista.DataObject.__getattr__` so the plotter
        and dataset extension points present the same lookup contract.
        """
        # Lazy import to avoid a circular dependency at module load time.
        from pyvista.plotting.component_registry import _resolve_pending_component  # noqa: PLC0415
    
        if _resolve_pending_component(item):
            return object.__getattribute__(self, item)
>       return super().__getattribute__(item)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       AttributeError: 'Plotter' object has no attribute '_picking_right_clicking_observer'

/__t/Python/3.12.13/x64/lib/python3.12/site-packages/pyvista/plotting/plotter.py:499: AttributeError
```

Close #4120

--- a/doc/changelog.d/4584.fixed.md
+++ b/doc/changelog.d/4584.fixed.md
@@ -1,0 +1,1 @@
+Update pyvista compatibility for picking observer in plotting functions

--- a/src/ansys/mapdl/core/plotting/plotting_defaults.py
+++ b/src/ansys/mapdl/core/plotting/plotting_defaults.py
@@ -21,7 +21,12 @@
 # SOFTWARE.
 
 import pyvista as pv
-from pyvista.core import _vtk_core as _vtk
+
+try:
+    from pyvista.core import _vtk_core as _vtk
+except ImportError:
+    from pyvista import _vtk  # pyvista >= 0.48
+
 from pyvista.core.utilities import translate
 from pyvista.core.utilities.helpers import wrap
 from pyvista.core.utilities.misc import _NoNewAttrMixin

--- a/tests/test_plotting.py
+++ b/tests/test_plotting.py
@@ -46,6 +46,17 @@ ALL_LABELS = FORCE_LABELS.copy()
 ALL_LABELS.extend(DISPL_LABELS)
 
 
+def _get_picking_right_clicking_observer(pl):
+    """Return the right-clicking observer, compatible with pyvista < and >= 0.48."""
+    if hasattr(pl, "_picking_right_clicking_observer"):
+        return pl._picking_right_clicking_observer
+    if hasattr(pl, "picking") and hasattr(
+        pl.picking, "_picking_right_clicking_observer"
+    ):
+        return pl.picking._picking_right_clicking_observer
+    return None
+
+
 @pytest.fixture
 def boundary_conditions_example(mapdl, cleared):
     mapdl.et("", 189)
@@ -566,7 +577,7 @@ def test_pick_nodes(mapdl, make_block, selection, verify_image_cache):
         pl.show(auto_close=False)
         pl.window_size = (100, 100)
         width, height = pl.window_size
-        if pl._picking_right_clicking_observer is None:
+        if _get_picking_right_clicking_observer(pl) is None:
             pl.iren._mouse_left_button_press(
                 int(width * point[0]), int(height * point[1])
             )
@@ -639,7 +650,7 @@ def test_pick_kp(mapdl, make_block, selection):
         pl.show(auto_close=False)
         pl.window_size = (100, 100)
         width, height = pl.window_size
-        if pl._picking_right_clicking_observer is None:
+        if _get_picking_right_clicking_observer(pl) is None:
             pl.iren._mouse_left_button_press(
                 int(width * point[0]), int(height * point[1])
             )
@@ -848,7 +859,7 @@ def test_pick_areas(mapdl, make_block, selection):
         pl.show(auto_close=False)
         pl.window_size = (100, 100)
         width, height = pl.window_size
-        if pl._picking_right_clicking_observer is None:
+        if _get_picking_right_clicking_observer(pl) is None:
             pl.iren._mouse_left_button_press(
                 int(width * point[0]), int(height * point[1])
             )

--- a/tests/test_plotting.py
+++ b/tests/test_plotting.py
@@ -47,7 +47,7 @@ ALL_LABELS.extend(DISPL_LABELS)
 
 
 def _get_picking_right_clicking_observer(pl):
-    """Return the right-clicking observer, compatible with pyvista < and >= 0.48."""
+    """Return the right-clicking observer, compatible with PyVista <0.48 and >=0.48."""
     if hasattr(pl, "_picking_right_clicking_observer"):
         return pl._picking_right_clicking_observer
     if hasattr(pl, "picking") and hasattr(

